src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public abstract class Figure implements Area, Drawing {
+    private String color;
+
+    public Figure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        this.color = colorSupplier.getRandomColor();
```

Comment:
A constructor must not have any business logic. It has only one responsibility - Object Initialization 
Please, pass all needed fields to parameters ;)